<?xml version="1.0" ?>
<sdf version="1.6">
  <world name="multi_box">
    <!-- A global light source -->
    <include>
      <uri>model://sun</uri>
    </include>
    <!-- A ground plane -->
    <include>
      <uri>model://ground_plane</uri>
    </include>
    <model name='logo_ground'>
      <include>
        <uri>model://logo_ground</uri>
	      <pose> 0 0 0.01 0 0 0</pose>
      </include>
    </model>
    <model name="black_box">
      <pose>1 0 0.2 0 0 0</pose>
      <static>true</static>
      <link name="link">
          <inertial>
            <mass>1.0</mass>
          </inertial>
          <collision name="my_collision">
            <geometry>
              <box>
                <size>0.5 0.5 0.1</size>
              </box>
            </geometry>
          </collision>
          <visual name="my_visual">
            <geometry>
              <box>
                <size>0.5 0.5 0.1</size>
              </box>
            </geometry>
            <material>
              <script>
                <name>Gazebo/Black</name>
                <uri>file://media/materials/scripts/gazebo.material</uri>
              </script>
            </material>
          </visual>
      </link>
    </model>
    <model name="red_box">
      <pose>2 0 0.2 0 0 0</pose>
      <static>true</static>
      <link name="link">
          <inertial>
            <mass>1.0</mass>
          </inertial>
          <collision name="my_collision">
            <geometry>
              <box>
                <size>0.5 0.5 0.1</size>
              </box>
            </geometry>
          </collision>
          <visual name="my_visual">
            <geometry>
              <box>
                <size>0.5 0.5 0.1</size>
              </box>
            </geometry>
            <material>
              <script>
                <name>Gazebo/Red</name>
                <uri>file://media/materials/scripts/gazebo.material</uri>
              </script>
            </material>
          </visual>
      </link>
    </model>
    <model name="yellow_box">
      <pose>3 0 0.2 0 0 0</pose>
      <static>true</static>
      <link name="link">
          <inertial>
            <mass>1.0</mass>
          </inertial>
          <collision name="my_collision">
            <geometry>
              <box>
                <size>0.5 0.5 0.1</size>
              </box>
            </geometry>
          </collision>
          <visual name="my_visual">
            <geometry>
              <box>
                <size>0.5 0.5 0.1</size>
              </box>
            </geometry>
            <material>
              <script>
                <name>Gazebo/Yellow</name>
                <uri>file://media/materials/scripts/gazebo.material</uri>
              </script>
            </material>
          </visual>
      </link>
    </model>
    <model name="blue_box">
      <pose>4 0 0.2 0 0 0</pose>
      <static>true</static>
      <link name="link">
          <inertial>
            <mass>1.0</mass>
          </inertial>
          <collision name="my_collision">
            <geometry>
              <box>
                <size>0.5 0.5 0.1</size>
              </box>
            </geometry>
          </collision>
          <visual name="my_visual">
            <geometry>
              <box>
                <size>0.5 0.5 0.1</size>
              </box>
            </geometry>
            <material>
              <script>
                <name>Gazebo/Blue</name>
                <uri>file://media/materials/scripts/gazebo.material</uri>
              </script>
            </material>
          </visual>
      </link>
    </model>

    <physics name='default_physics' default='0' type='ode'>
      <gravity>0 0 -9.8066</gravity>
      <ode>
        <solver>
          <type>quick</type>
          <iters>150</iters>
          <sor>1.3</sor>
          <use_dynamic_moi_rescaling>0</use_dynamic_moi_rescaling>
        </solver>
        <constraints>
          <cfm>0.00001</cfm>
          <erp>0.2</erp>
          <contact_max_correcting_vel>1000</contact_max_correcting_vel>
          <contact_surface_layer>0.01</contact_surface_layer>
        </constraints>
      </ode>
      <max_step_size>0.004</max_step_size>
      <real_time_factor>1000.0</real_time_factor>
      <real_time_update_rate>250</real_time_update_rate>
      <magnetic_field>6.0e-6 2.3e-5 -4.2e-5</magnetic_field>
    </physics>
  </world>
</sdf>

